moment_kinetics/src/load_data.jl

```diff
@@ -246,7 +246,9 @@ function load_coordinate_data(fid, name; printout=false, irank=nothing, nrank=no
         return nothing, nothing, nothing
     end
 
+    input = OptionsDict()
```

Comment:
I'm not sure I'm following your concern. Within `define_coordinate()`, the `input` is passed through `set_defaults_and_check_section()`, so any parameter that wouldn't be used would cause an error, and any parameter that is not set will use its default value. It is possible that `load_coordinate_data()` could be made simpler and/or more robust, I haven't given it any thought yet (and am not planning to any time soon).

Reply:
My concern is that one could imagine adding an extra parameter to the input parameter deck inside the call to `set_defaults_and_check_section()`, but not add the appropriate `input["new_parameter"] = ` here in load_data, and thus only every get the default value for "new_parameter", without getting an error. If we had to pass "new_parameter" to a struct first, we would get an error if we forgot to do it.

Reply:
I think the chances of making mistakes actually go down with the new approach.

With the old approach, `grid_input` was created with positional arguments, which are prone to errors (e.g. being entered in the wrong order) - that problem could be fixed using `@kwdef`, but actually you can get that effect with the new structure if you want, by creating a `NamedTuple` of inputs manually, which would cause errors if any input was missing. I'm not convinced that's useful though, unless you object to having default values at all.

In the new approach, if the inputs change then either:
* an input is removed, and the test uses that input - in this case, there will be an error message saying that input does not exist if it is not updated.
* an input is renamed - again you get a helpful error
* a new input is added - existing tests should not depend on this input, and should keep working using its default value. If the default is not OK, then the tests should fail unless we are very unlucky, and if you just added a new input parameter it shouldn't be too hard to realise that the new parameter needs to be set.

Reply:
OK, these seem like convincing arguments.